x/ibc/light-clients/solomachine/types/proposal_handle.go

```diff
@@ -0,0 +1,32 @@
+package types
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	clienttypes "github.com/cosmos/cosmos-sdk/x/ibc/02-client/types"
+	"github.com/cosmos/cosmos-sdk/x/ibc/exported"
+)
+
+// CheckProposedHeaderAndUpdateState updates the consensus state to the header's sequence and
+// public key. An error is returned if the header cannot be casted to a solo machine header.
+func (cs ClientState) CheckProposedHeaderAndUpdateState(
+	ctx sdk.Context, cdc codec.BinaryMarshaler, clientStore sdk.KVStore,
+	header exported.Header,
+) (exported.ClientState, exported.ConsensusState, error) {
+	smHeader, ok := header.(*Header)
+	if !ok {
+		return nil, nil, sdkerrors.Wrapf(
+			clienttypes.ErrInvalidHeader, "header type %T, expected  %T", header, &Header{},
+		)
+	}
+
+	consensusState := &ConsensusState{
+		Sequence:  smHeader.Sequence,
+		PublicKey: smHeader.NewPublicKey,
```

Comment:
Should we check that `NewPublicKey != CurrentPublicKey`?

Reply:
seems reasonable to me

Reply:
should this be checked on regular updates as well? cc @cwgoes

Reply:
on a related note, should the sequence set in consensus state == header.GetSequence() or should it be header.GetSequence + 1 like in `update.go`?

Reply:
I think we can check that `newPublicKey != currentPublicKey`, there's no reason to pass a proposal if you have access to the current public key.

This is a bit of a special case, since we're resetting the sequence, I think having it be equal is fine.